community/kernel/src/main/java/org/neo4j/kernel/impl/index/schema/ZonedTimeSchemaKey.java

```diff
@@ -82,9 +90,9 @@ public String toString()
     }
 
     @Override
-    public void writeTime( long nanosOfDayUTC, int offsetSeconds )
+    public void writeTime( long nanosOfDayLocal, int offsetSeconds )
     {
-        this.nanosOfDayUTC = nanosOfDayUTC;
+        this.nanosOfDayUTC = nanosOfDayLocal - offsetSeconds * 1_000_000_000L;
```

Comment:
Perhaps you want to use your new NANOS_PER_SECOND here instead of hardcode 1_000_000_000L?

Reply:
yupp.